processor/src/main/java/org/mapstruct/ap/processor/MethodRetrievalProcessor.java

```diff
@@ -80,19 +93,60 @@ public int getPriority() {
         return 1;
     }
 
+    private List<SourceMethod> retrievePrototypeMethods(TypeMirror typeMirror, MapperConfig mapperConfig) {
+        if ( typeMirror == null || typeMirror.getKind() == TypeKind.VOID ) {
+            return Collections.emptyList();
+        }
+
+        TypeElement typeElement = asTypeElement( typeMirror );
+        List<SourceMethod> methods = new ArrayList<SourceMethod>();
+        for ( ExecutableElement executable : getAllEnclosedExecutableElements( elementUtils, typeElement ) ) {
+
+            ExecutableType methodType = typeFactory.getMethodType( typeElement, executable );
+            List<Parameter> parameters = typeFactory.getParameters( methodType, executable );
+            boolean containsTargetTypeParameter = SourceMethod.containsTargetTypeParameter( parameters );
+
+            // prototype methods don't have prototypes themselves
```

Comment:
> The config I wish to make inheritable refers to an abstract return type.

What I meant is that a return type does not have to be abstract to be in an inheritance relation. The only way to have inheritance relation between return types in mapping method is as the base return type is in a mapping method in the `MappingConfig` class AFAIK.


> I'm not sure what you mean. Could you give an example for this case? A user can declare (and configure) e.g. an element mapping method which then is used by an iterable mapping method. Why would they declare that method on a config class rather than a mapper?

Of course that's possible (to be honest, I realized that later as well). But, perhaps the similar reason as you state above. To render a direct mapping invisible to the user. The generated forged method could then be private. If you map a object hierarchy, its kind of strange that you have to define all branch and leaves in an interface while you only require a mapping method at root level.